### ISSUE TITLE
Swagger model definitions are complicated and hard to follow

### ISSUE THREAD

**Issue by doomsbuster** [grafeas/grafeas#96]
The API definition have some room for improvement in that the model definitions should be concise and relatable. Current long list of metadata model is hard to relate to REST API request bodies and the hierarchy of models.

E.g. `apiBuildDetails` and `apiBuildProvenance` the API documentation is very hard to read, follow and make sense when it comes to defining API request bodies.

**Comment by vtsao** [comment]
Hi @doomsbuster our source of truth for this API is defined using protos, the Swagger spec is generated using the protoc tool with the Swagger option, which has some issues right now. See https://github.com/grafeas/grafeas/issues/379.

If you have specific suggestions on how to improve our schema for the different [note/occurrence protos](https://github.com/grafeas/grafeas/tree/master/proto/v1), feel free to submit proposals or file seperate issues.

Otherwise I'm closing this as it is too vague/general.
